tron/serialize/runstate/statemanager.py

```diff
@@ -161,7 +161,11 @@ def restore(self, job_names, skip_validation=False):
                 for job_name, job_state in jobs.items()
             }
             for result in concurrent.futures.as_completed(results):
-                jobs[results[result]]["runs"] = result.result()
+                try:
+                    jobs[results[result]]["runs"] = result.result()
+                except Exception as e:
+                    log.error(f"Failed to restore runs for {results[result]}: {e}")
```

Comment:
something like `log.exception(f"Unable to restore state for {results[result]} - exiting to avoid corrupting data.")` might be a little clearer in logs